### ISSUE TITLE
should not remove semicolon before regex when fixing

### ISSUE THREAD

**Issue by JounQin** [eslint/eslint#7782]
**Tell us about your environment**

* **ESLint Version:** 3.12.2
* **Node Version:** 6.9.1
* **npm Version:** 4.0.2

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

``` js
const argv = require('yargs').argv

module.exports = {
  root: true,
  parser: 'babel-eslint',
  plugins: [
    'babel',
    'react',
    'flowtype',
    'standard',
    `vue${argv.fix ? 'fix' : ''}`
  ],
  extends: [
    'standard',
    'plugin:flowtype/recommended'
  ],
  env: {
    browser: true
  },
  globals: {
    __DEV__: false,
    __MOCK__: false,
    IMG_PATH_PREFIX: false
  },
  parserOptions: {
    ecmaVersion: 6,
    sourceType: 'module',
    ecmaFeatures: {
      jsx: true
    }
  },
  rules: {
    'array-bracket-spacing': 2,
    'babel/object-curly-spacing': 2,
    'computed-property-spacing': 2,
    'eol-last': 2,
    'generator-star-spacing': 2,
    'jsx-quotes': [
      2,
      'prefer-double'
    ],
    'max-depth': 2,
    'max-len': [
      2,
      120,
      2
    ],
    'max-nested-callbacks': 2,
    'max-params': [2, 5],
    'space-before-function-paren': [
      2,
      {
        anonymous: 'always',
        named: 'never'
      }
    ],
    'object-curly-spacing': 0,
    'react/jsx-uses-react': 2,
    'react/jsx-uses-vars': 2,
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

``` js
const img = new Image();

/^https?:\/\//i.test(url) && (img.crossOrigin = 'Anonymous')
```

**What did you expect to happen?**

Do not fix it with remove semicolon

**What actually happened? Please include the actual, raw output from ESLint.**

semicolon is removed then it will throw an error because it was no longer a valid js expression.

``` js
const img = new Image()

/^https?:\/\//i.test(url) && (img.crossOrigin = 'Anonymous')
```

**Comment by not-an-aardvark** [comment]
Thanks for the report. I can reproduce this issue.
